web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java

```diff
@@ -307,12 +309,65 @@ void shouldFilterOnToDateIfValue() {
             final var sickLeaves = List.of(expectedSickLeave, anotherSickLeave);
 
             final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
-                    null, null, null, TO_END_DATE);
+                    null, null, null, TO_END_DATE, Collections.emptyList());
 
             assertEquals(0, actualSickLeaveList.size());
         }
     }
 
+    @Nested
+    class FilterOnDoctorIds {
+        List<SjukfallEnhet> sickLeaves;
+        Lakare doctor1;
+        Lakare doctor2;
+
+        @BeforeEach
+        void setup() {
+            doctor1 = new Lakare();
+            doctor2 = new Lakare();
+            doctor1.setId("ID1");
+            doctor2.setId("ID2");
+
+            final var expectedSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 5, PATIENT_ID);
+            expectedSickLeave.setLakare(doctor1);
+            final var anotherSickLeave = createSjukFallEnhet(ANOTHER_DIAGNOSIS_CODE, 12, ANOTHER_PATIENT_ID);
+            anotherSickLeave.setLakare(doctor2);
+            sickLeaves = List.of(expectedSickLeave, anotherSickLeave);
+        }
+
+        @Test
+        void shouldNotFilterOnDoctorIdsIsNull() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, null);
+
+            assertEquals(sickLeaves, actualSickLeaveList);
+        }
+
+        @Test
+        void shouldNotFilterOnDoctorIdsIsEmpty() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.emptyList());
+
+            assertEquals(sickLeaves, actualSickLeaveList);
+        }
+
+        @Test
+        void shouldFilterOnDoctorId() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.singletonList(doctor1.getId()));
+
+            assertEquals(1, actualSickLeaveList.size());
+        }
+
+        @Test
+        void shouldNotFilterOnDoctorIds() {
```

Comment:
Consider changing the name of this test to something like.: shouldFilterOnMultipleDoctorIds